xchange-coingi/src/main/java/org/knowm/xchange/coingi/service/CoingiBaseService.java

```diff
@@ -0,0 +1,82 @@
+package org.knowm.xchange.coingi.service;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Optional;
+import org.knowm.xchange.Exchange;
+import org.knowm.xchange.coingi.dto.CoingiAuthenticatedRequest;
+import org.knowm.xchange.service.BaseExchangeService;
+import org.knowm.xchange.service.BaseService;
+import si.mazi.rescu.ClientConfig;
+import si.mazi.rescu.serialization.jackson.DefaultJacksonObjectMapperFactory;
+
+public class CoingiBaseService extends BaseExchangeService implements BaseService {
+  protected CoingiDigest signatureCreator;
+
+  protected CoingiBaseService(Exchange exchange) {
+    super(exchange);
+  }
+
+  @Override
+  public ClientConfig getClientConfig() {
+    ClientConfig config = super.getClientConfig();
+    config.setJacksonObjectMapperFactory(
+        new DefaultJacksonObjectMapperFactory() {
+          @Override
+          public ObjectMapper createObjectMapper() {
+            return super.createObjectMapper();
+          }
+
+          @Override
+          public void configureObjectMapper(ObjectMapper objectMapper) {
+            super.configureObjectMapper(objectMapper);
+            SimpleModule simpleModule = new SimpleModule();
+            simpleModule.addSerializer(
+                Optional.class, new CoingiBaseService.OptionalValueSerializer());
+            objectMapper.registerModule(simpleModule);
+          }
+        });
```

Comment:
I don't see where is this needed
the Optional class is not used in any DTO nor does it need to be
I think this plus the custom serialize should be removed
besides there is a standard Jackson module with that if we ever need it in the future

Reply:
done, thanks. Historically, Optional was used in the very first commit here. Since then, we've simplified code and managed to remove optional.